superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx

```diff
@@ -178,10 +174,21 @@ const FilterBar: React.FC<FiltersBarProps> = ({
 
   const handleFilterSelectionChange = (
     filter: Pick<Filter, 'id'> & Partial<Filter>,
-    dataMask: Partial<DataMaskState>,
+    dataMask: Partial<DataMask>,
   ) => {
     setIsFilterSetChanged(tab !== TabIds.AllFilters);
     setDataMaskSelected(draft => {
+      // force instant updating on initialization or for parent filters when dataMaskSelected has filter
```

Comment:
This comment is not fully accurate; can we update this to reflect the fact that we are updating on initialization if `filter.requiredFirst` is true?

Reply:
it happens not only on initialisation - it also happens when we update filter modal, because first it set as undefined and then get correct first value (meantime it still do double loading but need fix later who will have time)